src/nhs_faces.jl

```diff
@@ -0,0 +1,200 @@
+struct FaceNeighborhoodSearch{NDIMS, ELTYPE, PB}
+    cell_list         :: Dict{NTuple{NDIMS, Int}, Vector{Int}}
```

Comment:
Is there a reason why you cannot use existing cell lists here?

Reply:
No, when I started implementing this, PointNeighbors.jl was still living in TrixiParticles.jl and the NHS was not as mature as it is now. I'll switch to `FullGridCellList`

Reply:
However, I was not sure about `max_points_per_cell`. I mean there can be very large faces among very small faces, so the `max_points_per_cell` may vary very heavily.

Reply:
also it is very misleading to set max "points" here, as these are actually faces in this case

Reply:
Make the cell list a kwarg, which defaults to `DictionaryCellList`, just like the GNHS. Then the default always works, but users can explicitly pass a `FullGridCellList` and manually tweak `max_points_per_cell`.